main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala

```diff
@@ -230,6 +237,44 @@ object CodeActionProvider {
     command = None
   )
 
+  /**
+    * Returns a quickfix code action to derive the `Eq` type class for the given type `tpe` if it is an enum.
+    */
+  private def mkDeriveMissingEq(tpe: Type, uri: String): Option[CodeAction] =
+    mkDeriveMissing(tpe, "Eq", uri)
+
+  /**
+    * Returns a quickfix code action to derive the `Order` type class for the given type `tpe` if it is an enum.
+    */
+  private def mkDeriveMissingOrder(tpe: Type, uri: String): Option[CodeAction] =
+    mkDeriveMissing(tpe, "Order", uri)
+
+  /**
+    * Returns a quickfix code action to derive the `ToString` type class for the given type `tpe` if it is an enum.
+    */
+  private def mkDeriveMissingToString(tpe: Type, uri: String): Option[CodeAction] =
+    mkDeriveMissing(tpe, "ToString", uri)
+
+  /**
+    * Internal helper function for all `mkDeriveMissingX`.
+    * Returns a quickfix code action to derive the given type class `clazz` for the given type `tpe` if it is an enum.
+    */
+  private def mkDeriveMissing(tpe: Type, clazz: String, uri: String): Option[CodeAction] = tpe.typeConstructor match {
```

Comment:
Any particular reason not to reuse `mkDerive`?

Reply:
Just that one is a refactor and one is a quickfix.